client/src/components/CommentApp/utils/layout.ts

```diff
@@ -103,79 +105,97 @@ export class LayoutController {
       })
     );
 
-    // Sort blocks
-    blocks.sort(
-      (block, comparisonBlock) => block.position - comparisonBlock.position
-    );
-
-    // Resolve overlapping blocks
-    let overlaps = true;
-    while (overlaps) {
-      overlaps = false;
-      const newBlocks: Block[] = [];
-      let previousBlock: Block | null = null;
-      const pinnedCommentPosition = this.pinnedComment ?
-        this.commentDesiredPositions.get(this.pinnedComment) : undefined;
-
-      for (const block of blocks) {
-        if (previousBlock) {
-          if (
-            previousBlock.position + previousBlock.height + GAP >
-            block.position
-          ) {
-            overlaps = true;
-
-            // Merge the blocks
-            previousBlock.comments.push(...block.comments);
-
-            if (block.containsPinnedComment) {
-              previousBlock.containsPinnedComment = true;
-              previousBlock.pinnedCommentPosition =
-                block.pinnedCommentPosition + previousBlock.height;
-            }
-            previousBlock.height += block.height;
-
-            // Make sure comments don't disappear off the top of the page
-            // But only if a comment isn't focused
-            if (
-              !this.pinnedComment &&
-              previousBlock.position < TOP_MARGIN + OFFSET
-            ) {
-              previousBlock.position =
-                TOP_MARGIN + previousBlock.height - OFFSET;
-            }
+    // Group blocks by tabs
+    const blocksByTab: Map<string | null, Block[]> = new Map();
+    allBlocks.forEach(block => {
+      const blocks = blocksByTab.get(block.tab) || [];
+      blocks.push(block);
+      blocksByTab.set(block.tab, blocks);
+    });
 
-            // If this block contains the focused comment, position it so
-            // the focused comment is in it's desired position
+    // Get location of pinned comment
+    const pinnedCommentPosition = this.pinnedComment ?
+      this.commentDesiredPositions.get(this.pinnedComment) : undefined;
+    const pinnedCommentTab = this.pinnedComment ?
+      this.commentTabs.get(this.pinnedComment) : undefined;
+
+    // For each tab, resolve positions of all the comments
```

Comment:
Do we need to resolve positions on all tabs? Could we do just the current?

Reply:
Helps with initial page load to resolve them all, I didn't think it was worth optimising this much since it's pretty fast